File: src/main/java/com/arekalov/islab1/dto/FlatDTO.java
package com.arekalov.islab1.dto;

import com.arekalov.islab1.entity.Furnish;
import com.arekalov.islab1.entity.View;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.ZonedDateTime;

/**
 * DTO для передачи данных о квартире
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class FlatDTO {
    private Long id;
    private String name;
    private CoordinatesDTO coordinates;
    
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
    private ZonedDateTime creationDate;
    
    private Long area;
    private Long price;
    private Boolean balcony;
    private Long timeToMetroOnFoot;
    private Integer numberOfRooms;
    private Long livingSpace;
    private Furnish furnish;
    private View view;
    private HouseDTO house;
}

File: src/main/kotlin/com/arekalov/islab1/dto/FlatDTO.kt
package com.arekalov.islab1.dto

import com.arekalov.islab1.entity.Furnish
import com.arekalov.islab1.entity.View
import com.fasterxml.jackson.annotation.JsonFormat
import java.time.ZonedDateTime

/**
 * DTO для передачи данных о квартире
 */
data class FlatDTO(
    val id: Long?,
    val name: String,
    val coordinates: CoordinatesDTO,
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
    val creationDate: ZonedDateTime,
    val area: Long,
    val price: Long,
    val balcony: Boolean?,
    val timeToMetroOnFoot: Long,
    val numberOfRooms: Int,
    val livingSpace: Long,
    val furnish: Furnish,
    val view: View,
    val house: HouseDTO?
)

/**
 * DTO для координат
 */
data class CoordinatesDTO(
    val id: Long?,
    val x: Int,
    val y: Int
)

/**
 * DTO для дома
 */
data class HouseDTO(
    val id: Long?,
    val name: String?,
    val year: Int,
    val numberOfFlatsOnFloor: Int
)
